NAMING.md

```diff
@@ -0,0 +1,373 @@
+
+```js
+const SalesPerson = {
+  name: "SalesPerson",
+  pluralName: "SalesPeople"
+  attributes: {
+    firstName: "STRING",
+  },
+}
+```
+
+1. Creates a service layer `sales-people`.
+1. Creates relationships as `salesPerson` or `salesPeople` (`snakeCase` of `name` or `pluralName`).
+
+### Relationships
+
+Other than the required `target` field, you can leverage `as` to set the relationship name for service layer, URLs and database, or any of the below to control the naming in the database layer:
+
+- `foreignKey` to set the name of the foreign key. By default it is `${model}Id`
+- `sourceKey` to set the other table's foreign key column name for `hasOne` and `hasMany`
+- `targetKey` to set the other table's foreign key column name for `belongsTo` and `belongsToMany`
+- `through` is used for `belongsToMany` to control the name of the junction table
+
+For more information, refer to the [Sequelize Docs](https://sequelize.org/docs/v6/core-concepts/assocs/)
+
+### For `BelongsTo` relationships …
+
+`as` is used:
+
+```js
+const Account = {
+  name: "Account",
+  attributes: {
+    name: "STRING",
+  },
+  belongsTo: [{ as: "closerPerson", target: "SalesPerson" }],
+}
+```
+
+1. Creates a table `sales_person` with the columns `id` and `first_name` and a table `account` with the columns `id`, `name` and `closer_person_id`.
+1. Creates relationships as `closerPerson`.
+1. Returns `JSONAPI` from `GET /account/1?include=closer-person` like:
+
+```json
+{
+  "jsonapi": {
+    "version": "1.0"
+  },
+  "data": {
+    "type": "Account",
+    "id": "12-22",
+    "attributes": {
+      "name": "Bitovi"
+    },
+    "relationships": {
+      "closerPerson": {
+        "data": {
+          "type": "SalesPerson",
+          "id": "332-222"
+        }
+      }
+    },
+    "included": [
+      {
+        "type": "SalesPerson",
+        "id": "332-222",
+        "attributes": {
+          "firstName": "John"
+        }
+      }
+    ]
+  }
+}
+```
+
+`as` and `foreignKey` are used:
+
+```js
+const Account = {
+  name: "Account",
+  attributes: {
+    name: "STRING",
+  },
+  belongsTo: [{ as: "closer", foreignKey: "finisher_id", target: "SalesPerson" }],
+}
+```
+
+1. Creates a table `sales_person` with the columns `id` and `first_name` and a table `account` with the columns `id`, `name` and `finisher_id`.
+1. Creates relationships as `closerPerson`.
+1. Returns `JSONAPI` from `GET /account/1?include=closer` like:
+
+```json
+{
+  "jsonapi": {
+    "version": "1.0"
+  },
+  "data": {
+    "type": "Account",
+    "id": "12-22",
+    "attributes": {
+      "name": "Bitovi"
+    },
+    "relationships": {
+      "closer": {
+        "data": {
+          "type": "SalesPerson",
+          "id": "332-222"
+        }
+      }
+    },
+    "included": [
+      {
+        "type": "SalesPerson",
+        "id": "332-222",
+        "attributes": {
+          "firstName": "John"
+        }
+      }
+    ]
+  }
+}
+```
+
+### For `HasMany` relationships …
+
+`as` is used:
```

Comment:
Maybe somehow let it more clear that this is the behavior for when **only** "as" is used?